File: Practice/10/C++/10. Фиксированная сумма/10. Фиксированная сумма/10. Фиксированная сумма.cpp
#include <iostream>

int main()
{
        int l1, l2, r1, r2, S;
        setlocale(LC_ALL, "Ru");
        std::cout << "Введите целые числа через пробел \n S l1 l2 r1 r2" << std::endl;
        std::cin >> S >> l1 >> l2 >> r1 >> r2;
        if (l1 + r2 >= S) {
            r2 = std::max(S - l1, l2);
        }
        else {
            l1 = std::min(S - r2, r1);
        }
        if (l1 + r2 == S) {
            std::cout << l1 << " " << r2 << std::endl;
        }
        else {
            std::cout << -1 << std::endl;
        }
    

}

File: Practice/10/Python/10. Фиксированная сумма/10. Фиксированная сумма/_10._Фиксированная_сумма.py
data = input("Введите данные в таком порядке s, l1, r1, l2, r2 через пробел\n")
try:
    s, l1, r1, l2, r2 = [int(i) for i in data.split(" ")]

    if l1 + r2 >= s:
        r2 = max(s-l1, l2)
    else:
        l1 = min(s-r2, r1)

    print((l1, r2) if l1 + r2 == s else -1)
except ValueError:
    print("Ошибка")